internal/jobcontainers/jobcontainer.go

```diff
@@ -447,3 +525,63 @@ func (c *JobContainer) IsOCI() bool {
 func (c *JobContainer) OS() string {
 	return "windows"
 }
+
+// For every process in the job `job`, run the function `work`. This can be used to grab/filter the SYSTEM_PROCESS_INFORMATION
+// data from every process in a job.
+func forEachProcessInfo(job *jobobject.JobObject, work func(*winapi.SYSTEM_PROCESS_INFORMATION)) error {
+	procInfos, err := systemProcessInformation()
+	if err != nil {
+		return err
+	}
+
+	pids, err := job.Pids()
+	if err != nil {
+		return err
+	}
+
+	pidsMap := make(map[uint32]struct{})
+	for _, pid := range pids {
+		pidsMap[pid] = struct{}{}
+	}
+
+	for _, procInfo := range procInfos {
+		if _, ok := pidsMap[uint32(procInfo.UniqueProcessID)]; ok {
+			work(procInfo)
+		}
+	}
+	return nil
+}
+
+// Get a slice of SYSTEM_PROCESS_INFORMATION for all of the processes running on the system.
+func systemProcessInformation() ([]*winapi.SYSTEM_PROCESS_INFORMATION, error) {
+	var (
+		systemProcInfo *winapi.SYSTEM_PROCESS_INFORMATION
+		procInfos      []*winapi.SYSTEM_PROCESS_INFORMATION
+		size           = uint32(1024 * 512)
```

Comment:
What is this size from?

Reply:
Buffer size they use in hcs, could either be larger or smaller really it's just a nice default. If the buffers big enough to hold the data on one call we can skip subsequent ones.

Reply:
On my devbox if the buffers 1MB I've only ever seen 1 call but I think I'll stick to this default.

Reply:
Want me to make a comment here to the tune of 
`This happens to be the buffer size hcs uses but there's no really no hard need to keep it the same, it's just a sane default.`

Reply:
Eh I don't think that's necessary imo since we grow the buffer to the right size anyways